docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,139 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need new api that will allow us to skip cluster/host validations.
+This api should have it's own feature flag to allow us to disable it.
+
+It should be possible to use this API in an unsupported way, it is currently undecided whether or not this will be open or not.
+An event should be emitted whenever the API is used during an installation, this can be used to drive signatures etc.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
```

Comment:
Are there mandatory validations that cannot be skipped?

Reply:
Good question, from what I know of the motivations, it would seem that this feature is for the purpose of bypassing issues with specific validations. I question whether or not we should be skipping anything that is not known to be problematic by the user.

This would go against the idea of skipping all validations and instead favour skipping specific validations.

Reply:
Not sure regarding this, i would not start allowing to do override in UI and if someone is doing it on purpose, it's sort of his problem. Lets set relevant message, flag in jira, event and skip those failures from our statistics

Reply:
I didn't mean to suggest that anything should be added to the UI to allow this. Simply that when a cluster has overridden validations that we may want to display something to reflect this.

Reply:
Moreover, to clarify further, the only means by which a validation may be skipped is via this limited API. As agreed upon elsewhere, this will only occur for specific customers who will be aware of the consequences of using the feature. (namely that the cluster is unsupported.)